### ISSUE TITLE
Request at missing url problem

### ISSUE THREAD

**Issue by tomy2105** [elsa-workflows/elsa-core#1253]
Elsa 2.1 freshly baked/compiled.....

When you issue a HTTP request at a URL which is not handled by any workflow you get following exception
```
fail: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[1]
      An unhandled exception has occurred while executing the request.
      System.InvalidOperationException: The request reached the end of the pipeline without executing the endpoint: '/_Host'. Please register the EndpointMiddleware using 'IApplicationBuilder.UseEndpoints(...)' if using routing.
         at Microsoft.AspNetCore.Builder.ApplicationBuilder.<>c.<Build>b__18_0(HttpContext context)
         at Elsa.Activities.Http.Middleware.HttpEndpointMiddleware.InvokeAsync(HttpContext httpContext, IWorkflowLaunchpad workflowLaunchpad, IWorkflowInstanceStore workflowInstanceStore, IWorkflowRegistry workflowRegistry, IWorkflowBlueprintReflector workflowBlueprintReflector, IEnumerable`1 contentParsers) in D:\Projects\Envox\elsa-core\src\activities\Elsa.Activities.Http\Middleware\HttpEndpointMiddleware.cs:line 51
         at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.Invoke(HttpContext context)
         at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
         at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
         at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
         at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```
In addition this exception is propagated out in form of 500 response.

In Elsa 2.0 in such scenario there was no exception and it just answered with 404 (not available) without content....

**Comment by sfmskywalker** [comment]
I just tried this with the Elsa.Samples.Server.Host project, but there it seems to work fine - I am getting a 404 when I try e.g. `https://localhost:11000/workflows/test2`.

Could it be that you have your own startup without having registered the `EndpointMiddleware`?
Or are you seeing this in another sample project?

**Comment by tomy2105** [comment]
Nope got this using Elsa sample project ElsaDashboard.Samples.AspNetCore.Monolith.
And as you can see HttpEndpointMiddleware is in the stack trace of exception so it is obviously registered :(.

I grabed fresh source from git right now, compiled npm, compiled vs solution, deleted sqlite file so no workflows exist and did: https://localhost:15265/workflows/test123
And it did it again :).

**Comment by sfmskywalker** [comment]
> And as you can see HttpEndpointMiddleware is in the stack trace of exception so it is obviously registered

Sure, but I literally meant ASP.NET Core's `EndpointMiddleware` :)

Anyway, that had nothing to do with it either, you were right about the issue in the monolith sample project.
Turns out that the issue happens when using `app.Map` to branch middleware out - when you do, it does not merge back into the main branch, which is the issue.

There's an `app.UseWhen`, but I figured I might just as well implement the path check myself within `HttpEndpointMiddleware`. I tried and that works, so will push an update soon.

I also noticed however that when you try to request a workflow that doesn't exist, it will ultimately fallback to the _Host.cshtml page. Which makes sense, because that's how the pipeline is configured.

But I think that if you configure the HTTP activities module with a path prefix like `/workflows`, the intent is clearly to execute a workflow. So, if the workflow doesn't exist, it should return 404.

I'll be updating the code accordingly, but let me know if anyone has opinions about this.

@tomy2105 @vlad-crisan
